extensions/amp-analytics/0.1/activity-impl.js

```diff
@@ -124,7 +124,9 @@ const ACTIVE_EVENT_TYPES = [
  * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampDoc
  */
 export function installActivityServiceForTesting(ampDoc) {
-  registerServiceBuilderForDoc(ampDoc, 'activity', Activity);
+  registerServiceBuilderForDoc(ampDoc,
+      'activity',
+      ampdoc => new Activity(ampdoc));
```

Comment:
Apologies for being unclear - this is what I meant as well - we'd use constructors directly as factories.